Failed to activate the atom-typescript package

[Enter steps to reproduce below:]

1. ...
2. ...

**Atom Version**: 1.2.4
**System**: Ubuntu 14.04.3
**Thrown From**: [atom-typescript](https://github.com/TypeStrong/atom-typescript) package, v7.12.0


### Stack Trace

Failed to activate the atom-typescript package

```
At Cannot find module './helper'

Error: Cannot find module './helper'
    at Module._resolveFilename (module.js:336:15)
    at Function.Module._resolveFilename (/usr/share/atom/resources/app.asar/src/module-cache.js:383:52)
    at Function.Module._load (module.js:286:25)
    at Module.require (module.js:365:17)
    at require (module.js:384:17)
    at Object.<anonymous> (/home/reiser/.atom/packages/atom-typescript/node_modules/atom-package-deps/lib/main.js:7:15)
    at Module._compile (module.js:434:26)
    at Object.defineProperty.value [as .js] (/usr/share/atom/resources/app.asar/src/compile-cache.js:190:21)
    at Module.load (module.js:355:32)
    at Function.Module._load (module.js:310:12)
    at Module.require (module.js:365:17)
    at require (module.js:384:17)
    at Object.activate (/home/reiser/.atom/packages/atom-typescript/dist/main/atomts.js:148:5)
    at Package.module.exports.Package.activateNow (/usr/share/atom/resources/app.asar/src/package.js:170:19)
    at /usr/share/atom/resources/app.asar/src/package.js:150:32
    at Package.module.exports.Package.measure (/usr/share/atom/resources/app.asar/src/package.js:92:15)
    at /usr/share/atom/resources/app.asar/src/package.js:143:26
    at Package.module.exports.Package.activate (/usr/share/atom/resources/app.asar/src/package.js:140:34)
    at PackageManager.module.exports.PackageManager.activatePackage (/usr/share/atom/resources/app.asar/src/package-manager.js:512:21)
    at /usr/share/atom/resources/app.asar/node_modules/settings-view/lib/package-manager.js:444:29
    at exit (/usr/share/atom/resources/app.asar/node_modules/settings-view/lib/package-manager.js:73:16)
    at triggerExitCallback (/usr/share/atom/resources/app.asar/src/buffered-process.js:213:47)
    at ChildProcess.<anonymous> (/usr/share/atom/resources/app.asar/src/buffered-process.js:235:18)
    at emitTwo (events.js:87:13)
    at ChildProcess.emit (events.js:172:7)
    at Process.ChildProcess._handle.onexit (internal/child_process.js:200:12)
```

### Commands

```
     -9:12.2.0 settings-view:open (atom-pane.pane.active)
 15x -8:35.8.0 core:backspace (atom-text-editor.editor.mini.is-focused)
     -7:42.4.0 core:cancel (div.package-detail.panels-item)
  8x -1:10.9.0 core:backspace (atom-text-editor.editor.mini.is-focused)
     -1:01.4.0 core:confirm (atom-text-editor.editor.mini.is-focused)
     -0:32.1.0 core:backspace (atom-text-editor.editor.mini.is-focused)
     -0:31.6.0 core:confirm (atom-text-editor.editor.mini.is-focused)
```

### Config

```json
{
  "core": {},
  "atom-typescript": {}
}
```

### Installed Packages

```coffee
# User
atom-typescript, vundefined

# Dev
No dev packages
```

I am getting this same error on Arch Linux. Namely this is my full error report:


[Enter steps to reproduce below:]

1. Edit Atom's package.json to include this package and build Atom
2. Start Atom

**Atom Version**: 1.7.4
**System**: "Arch Linux"
**Thrown From**: [atom-typescript](https://github.com/TypeStrong/atom-typescript) package, v8.10.2


### Stack Trace

Failed to activate the atom-typescript package

```
At Cannot find module '../../../../views/renameView.html'

Error: Cannot find module '../../../../views/renameView.html'
    at Module._resolveFilename (module.js:338:15)
    at Function.Module._resolveFilename (/usr/share/atom/resources/app.asar/src/module-cache.js:383:52)
    at Function.Module._load (module.js:289:25)
    at Module.require (module.js:366:17)
    at require (/usr/share/atom/resources/app.asar/src/native-compile-cache.js:50:27)
    at Object.<anonymous> (/usr/share/atom/resources/app.asar/node_modules/atom-typescript/dist/main/atom/views/renameView.js:9:12)
    at Module._compile (/usr/share/atom/resources/app.asar/src/native-compile-cache.js:103:30)
    at Object.defineProperty.value [as .js] (/usr/share/atom/resources/app.asar/src/compile-cache.js:208:21)
    at Module.load (module.js:356:32)
    at Function.Module._load (module.js:313:12)
    at Module.require (module.js:366:17)
    at require (/usr/share/atom/resources/app.asar/src/native-compile-cache.js:50:27)
    at Object.<anonymous> (/usr/share/atom/resources/app.asar/node_modules/atom-typescript/dist/main/atom/commands/commands.js:10:18)
    at Module._compile (/usr/share/atom/resources/app.asar/src/native-compile-cache.js:103:30)
    at Object.defineProperty.value [as .js] (/usr/share/atom/resources/app.asar/src/compile-cache.js:208:21)
    at Module.load (module.js:356:32)
    at Function.Module._load (module.js:313:12)
    at Module.require (module.js:366:17)
    at require (/usr/share/atom/resources/app.asar/src/native-compile-cache.js:50:27)
    at Object.<anonymous> (/usr/share/atom/resources/app.asar/node_modules/atom-typescript/dist/main/atomts.js:11:16)
    at Module._compile (/usr/share/atom/resources/app.asar/src/native-compile-cache.js:103:30)
    at Object.defineProperty.value [as .js] (/usr/share/atom/resources/app.asar/src/compile-cache.js:208:21)
    at Module.load (module.js:356:32)
    at Function.Module._load (module.js:313:12)
    at Module.require (module.js:366:17)
    at require (/usr/share/atom/resources/app.asar/src/native-compile-cache.js:50:27)
    at Package.module.exports.Package.requireMainModule (/usr/share/atom/resources/app.asar/src/package.js:715:27)
    at Package.module.exports.Package.activateNow (/usr/share/atom/resources/app.asar/src/package.js:173:16)
    at /usr/share/atom/resources/app.asar/src/package.js:156:32
    at Package.module.exports.Package.measure (/usr/share/atom/resources/app.asar/src/package.js:92:15)
```

### Commands

```
```

### Config

```json
{
  "core": {
    "customFileTypes": {
      "source.ini": [
        ".buckconfig",
        ".flowconfig"
      ],
      "source.json": [
        ".arcconfig",
        "BUCK.autodeps"
      ],
      "source.lisp": [
        ".emacs",
        ".spacemacs"
      ],
      "source.python": [
        "BUCK"
      ],
      "text.html.basic": [
        "ejs",
        "htm",
        "html",
        "kit",
        "shtml",
        "tmpl",
        "tpl",
        "xhtml"
      ]
    },
    "disabledPackages": [
      "language-markdown",
      "atom-beautify",
      "atomic-chrome",
      "build",
      "Hydrogen",
      "git-plus",
      "language-asciidoc"
    ],
    "projectHome": "/home/fusion809/GitHub/mine",
    "themes": [
      "fusion-ui",
      "dark-bint-syntax"
    ]
  }
}
```

### Installed Packages

```coffee
# User
No installed packages

# Dev
No dev packages
```

I'm seeing this error as well
[Enter steps to reproduce below:]

1. Enable package
2. Open Atom

**Atom Version**: 1.10.2
**Electron Version**: 0.37.8
**System**: Mac OS X 10.11.6
**Thrown From**: [atom-typescript](https://github.com/TypeStrong/atom-typescript) package, v10.1.7


### Stack Trace

Failed to activate the atom-typescript package

```
At Cannot find module 'atom-package-deps'

Error: Cannot find module 'atom-package-deps'
    at Module._resolveFilename (module.js:339:15)
    at Function.Module._resolveFilename (/opt/homebrew-cask/Caskroom/atom/1.3.3/Atom.app/Contents/Resources/app.asar/src/module-cache.js:383:52)
    at Function.Module._load (module.js:290:25)
    at Module.require (module.js:367:17)
    at require (/opt/homebrew-cask/Caskroom/atom/1.3.3/Atom.app/Contents/Resources/app.asar/src/native-compile-cache.js:50:27)
    at Object.activate (/Users/yolk/.atom/packages/atom-typescript/dist/main/atomts.js:149:5)
    at Package.module.exports.Package.activateNow (/opt/homebrew-cask/Caskroom/atom/1.3.3/Atom.app/Contents/Resources/app.asar/src/package.js:183:20)
    at /opt/homebrew-cask/Caskroom/atom/1.3.3/Atom.app/Contents/Resources/app.asar/src/package.js:156:32
    at Package.module.exports.Package.measure (/opt/homebrew-cask/Caskroom/atom/1.3.3/Atom.app/Contents/Resources/app.asar/src/package.js:92:15)
    at /opt/homebrew-cask/Caskroom/atom/1.3.3/Atom.app/Contents/Resources/app.asar/src/package.js:149:26
    at Package.module.exports.Package.activate (/opt/homebrew-cask/Caskroom/atom/1.3.3/Atom.app/Contents/Resources/app.asar/src/package.js:146:34)
    at PackageManager.module.exports.PackageManager.activatePackage (/opt/homebrew-cask/Caskroom/atom/1.3.3/Atom.app/Contents/Resources/app.asar/src/package-manager.js:538:21)
    at /opt/homebrew-cask/Caskroom/atom/1.3.3/Atom.app/Contents/Resources/app.asar/src/package-manager.js:519:29
    at Config.module.exports.Config.transactAsync (/opt/homebrew-cask/Caskroom/atom/1.3.3/Atom.app/Contents/Resources/app.asar/src/config.js:337:18)
    at PackageManager.module.exports.PackageManager.activatePackages (/opt/homebrew-cask/Caskroom/atom/1.3.3/Atom.app/Contents/Resources/app.asar/src/package-manager.js:514:19)
    at PackageManager.module.exports.PackageManager.activate (/opt/homebrew-cask/Caskroom/atom/1.3.3/Atom.app/Contents/Resources/app.asar/src/package-manager.js:497:46)
    at /opt/homebrew-cask/Caskroom/atom/1.3.3/Atom.app/Contents/Resources/app.asar/src/atom-environment.js:765:28
```

### Commands

```
```

### Config

```json
{
  "core": {
    "disabledPackages": [
      "jake"
    ],
    "themes": [
      "atom-dark-ui",
      "enlightened"
    ]
  }
}
```

### Installed Packages

```coffee
# User
aligner, v0.22.2 (inactive)
atom-typescript, v10.1.7 (inactive)
base16-ocean-dark-spacegray-lighter-invisibles, v0.3.0 (inactive)
copy-path, v0.4.0 (inactive)
editorconfig, v1.4.1 (inactive)
enlightened, v0.2.6 (inactive)
expand-selection-to-quotes, v0.9.0 (inactive)
git-diff-details, v1.3.0 (inactive)
git-plus, v5.17.1 (inactive)
git-time-machine, v1.5.3 (inactive)
gruvbox-plus-syntax, v1.2.5 (inactive)
language-docker, v1.1.7 (inactive)
language-glsl, v2.0.1 (inactive)
language-groovy, v0.6.0 (inactive)
language-javascript-jsx, v0.3.7 (inactive)
language-powershell, v4.0.0 (inactive)
linter, v1.11.16 (inactive)
linter-coffeescript, v1.0.0 (inactive)
linter-js-yaml, v1.2.5 (inactive)
markdown-writer, v2.6.1 (inactive)
minimap, v4.25.0 (inactive)
minimap-git-diff, v4.3.1 (inactive)
minimap-highlight-selected, v4.4.0 (inactive)
minimap-pigments, v0.2.1 (inactive)
one-dark-atom, v1.3.0 (inactive)
one-dark-shade-ui, v0.1.0 (inactive)
one-dork-ui, v2.0.3 (inactive)
one-outlander-ui, v0.3.1 (inactive)
open-terminal-here, v2.1.4 (inactive)
set-syntax, v0.3.0 (inactive)
sync-settings, v0.7.2 (inactive)
tidy-markdown, v3.0.1 (inactive)
wakatime, v6.0.10 (inactive)
atom-dark-syntax, v0.27.0 (inactive)
atom-dark-ui, v0.52.0 (inactive)
atom-light-syntax, v0.28.0 (inactive)
atom-light-ui, v0.44.0 (inactive)
base16-tomorrow-dark-theme, v1.1.0 (inactive)
base16-tomorrow-light-theme, v1.1.1 (inactive)
one-dark-ui, v1.5.0 (inactive)
one-light-ui, v1.5.0 (inactive)
one-dark-syntax, v1.3.0 (inactive)
one-light-syntax, v1.3.0 (inactive)
solarized-dark-syntax, v1.0.2 (inactive)
solarized-light-syntax, v1.0.2 (inactive)
about, v1.5.3 (inactive)
archive-view, v0.61.1 (inactive)
autocomplete-atom-api, v0.10.0 (inactive)
autocomplete-css, v0.11.2 (inactive)
autocomplete-html, v0.7.2 (inactive)
autocomplete-plus, v2.31.1 (inactive)
autocomplete-snippets, v1.11.0 (inactive)
autoflow, v0.27.0 (inactive)
autosave, v0.23.1 (inactive)
background-tips, v0.26.1 (inactive)
bookmarks, v0.41.0 (inactive)
bracket-matcher, v0.82.1 (inactive)
command-palette, v0.38.0 (inactive)
deprecation-cop, v0.54.1 (inactive)
dev-live-reload, v0.47.0 (inactive)
encoding-selector, v0.22.0 (inactive)
exception-reporting, v0.39.0 (inactive)
find-and-replace, v0.201.0 (inactive)
fuzzy-finder, v1.3.0 (inactive)
git-diff, v1.1.0 (inactive)
go-to-line, v0.31.0 (inactive)
grammar-selector, v0.48.1 (inactive)
image-view, v0.58.2 (inactive)
incompatible-packages, v0.26.1 (inactive)
keybinding-resolver, v0.35.0 (inactive)
line-ending-selector, v0.5.0 (inactive)
link, v0.31.1 (inactive)
markdown-preview, v0.158.0 (inactive)
metrics, v0.53.1 (inactive)
notifications, v0.65.0 (inactive)
open-on-github, v1.2.0 (inactive)
package-generator, v1.0.0 (inactive)
settings-view, v0.242.2 (inactive)
snippets, v1.0.2 (inactive)
spell-check, v0.67.1 (inactive)
status-bar, v1.4.1 (inactive)
styleguide, v0.47.0 (inactive)
symbols-view, v0.113.0 (inactive)
tabs, v0.100.2 (inactive)
timecop, v0.33.2 (inactive)
tree-view, v0.208.2 (inactive)
update-package-dependencies, v0.10.0 (inactive)
welcome, v0.34.0 (inactive)
whitespace, v0.33.0 (inactive)
wrap-guide, v0.38.1 (inactive)
language-c, v0.52.1 (inactive)
language-clojure, v0.21.0 (inactive)
language-coffee-script, v0.47.2 (inactive)
language-csharp, v0.12.1 (inactive)
language-css, v0.37.1 (inactive)
language-gfm, v0.88.0 (inactive)
language-git, v0.15.0 (inactive)
language-go, v0.42.1 (inactive)
language-html, v0.45.1 (inactive)
language-hyperlink, v0.16.0 (inactive)
language-java, v0.23.0 (inactive)
language-javascript, v0.119.0 (inactive)
language-json, v0.18.2 (inactive)
language-less, v0.29.5 (inactive)
language-make, v0.22.2 (inactive)
language-mustache, v0.13.0 (inactive)
language-objective-c, v0.15.1 (inactive)
language-perl, v0.35.0 (inactive)
language-php, v0.37.2 (inactive)
language-property-list, v0.8.0 (inactive)
language-python, v0.45.0 (inactive)
language-ruby, v0.69.0 (inactive)
language-ruby-on-rails, v0.25.0 (inactive)
language-sass, v0.56.0 (inactive)
language-shellscript, v0.22.4 (inactive)
language-source, v0.9.0 (inactive)
language-sql, v0.23.0 (inactive)
language-text, v0.7.1 (inactive)
language-todo, v0.28.0 (inactive)
language-toml, v0.18.0 (inactive)
language-xml, v0.34.9 (inactive)
language-yaml, v0.26.0 (inactive)

# Dev
No dev packages
```

--- a/lib/typescript/makeTypeScriptGlobal.ts
+++ b/lib/typescript/makeTypeScriptGlobal.ts
@@ -1,33 +1,21 @@
-////////////////////////////////// MAGIC
-import vm = require('vm');
-import fs = require('fs');
-import os = require('os');
-import path = require('path');
+import {isAbsolute} from "path"
 
+// Debugging helper
 global.stack = function() {
     console.error((<any>new Error()).stack);
 }
 
-/** Makes the bundled typescript services global or (if passed in) a custom typescriptServices file */
-export function makeTsGlobal(typescriptServices?: string) {
-    var sandbox = {
-        // This is going to gather the ts module exports
-        ts: {},
-        console: console,
-        stack: global.stack,
-        require: require,
-        module: module,
-        process: process
-    };
-    vm.createContext(sandbox);
-
-    if (typescriptServices) {
-        vm.runInContext(fs.readFileSync(typescriptServices).toString(), sandbox);
-    }
-    else {
-        sandbox.ts = require('typescript');
+// Export Typescript as a global. Takes an optional full path to typescriptServices.js
+export function makeTsGlobal(typescriptPath?: string) {
+  if (typescriptPath) {
+    if (!isAbsolute(typescriptPath)) {
+      throw new Error(`Path to Typescript "${typescriptPath}" is not absolute`)
     }
 
-    // Finally export ts to the local global namespace
-    global.ts = sandbox.ts;
+    typescriptPath = typescriptPath.trim()
+  } else {
+    typescriptPath = "typescript"
+  }
+
+  global.ts = require(typescriptPath);
 }

